utilities/upgrade-service/install-upgrade.sh

```diff
@@ -39,11 +67,36 @@ rollback() {
   exit 1;
 }
 
-cleanup() {
-  rm -rf /tmp/managed-gitops
+# creates a kustomization.yaml file in the temp directory pointing to the manifests available in the upstream repo.
+function create_kustomization_init_file() {
+  echo "apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - https://github.com/redhat-appstudio/managed-gitops/manifests/base/crd/overlays/local-dev?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/manifests/base/gitops-namespace?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/manifests/base/gitops-service-argocd/base?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/manifests/base/postgresql-staging?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/appstudio-controller/config/default?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/backend/config/default?ref=$GIT_REVISION
+  - https://github.com/redhat-appstudio/managed-gitops/cluster-agent/config/default?ref=$GIT_REVISION" > ${TEMP_DIR}/kustomization.yaml
 }
 
-exit_if_binary_not_installed kubectl
+# Code execution starts here
+# create a temporary directory and do all the operations inside the directory.
+GIT_REVISION=main
+TEMP_DIR=$(mktemp -d -t managed-gitops-install-XXXXXXX)
```

Comment:
each time the script is run, it creates a temporary directory with `X` replaced by unique chars to guarentee uniquenes and gets deleted at the end of the script run. So multiple simultaneous runs are already isolated. Addding version/image/username would increase the dir name/path and still not guarentee uniqueness.

Reply:
I think what is happening is random characters are appended to the directory name instead of replacing `XXXXXXX`. For e.g. `managed-gitops-install-XXXXXXX.aQu5qLcn`. But that does make it unique. In this case, we would want to remove `XXXXXXX` as having it doesn't make much sense IMO.

Reply:
That's bit strange. The `managed-gitops-install-XXXXXXX` is meant be used as a template (`-t` option) and should not be having `X` on the generated directory name.
I am not able to reproduce this.
On Mac, the generated name is something like this `mktemp -d -t managed-gitops-install-XXXXXXX
managed-gitops-install-jcvxLF4`
On a Fedora, the generated name is comes something like this `mktemp -d -t managed-gitops-install-XXXXXXX
/tmp/managed-gitops-install-sSHsxhE`

Reply:
I am running on Mac, and it seems to be giving me a huge path under `/var/folders`

e.g. from my terminal
```
MacBook-Pro:folders ishitasequeira$ TEMP_DIR=$(mktemp -d -t managed-gitops-install-XXXXXXX)
MacBook-Pro:folders ishitasequeira$ echo $TEMP_DIR 
/var/folders/_v/<random UUID>/T/managed-gitops-install-XXXXXXX.QVoBWK4q
```

Reply:
I was able to reproduce the problem when I tried with the `mktemp` binary that is part of the OS. I was using a gnu compatible version of the `mktemp` binary earlier and couldn't reproduce this issue. Mac is treating the whole template as a prefix and not doing a replacement of the `X's`. The expected behaviour is mentioned in this link https://ss64.com/osx/mktemp.html and this tells that the X's should get replaced.

Reply:
I think the solution to this is to not use `-t` and insert an explicit path to the created file.

Example:
```
mktemp -d "${TMPDIR:-/tmp}/managed-gitops-install-XXXXXXX"
```

Reply:
Made this change